### ISSUE TITLE
Implement kmeans++ initializer

### ISSUE THREAD

**Issue by atecon** [atecon/kmeans#2]
**Add support for the alternative initializer which reduces the risk of empty clusters**

- https://en.wikipedia.org/wiki/K-means%2B%2B
- https://scikit-learn.org/stable/auto_examples/cluster/plot_mini_batch_kmeans.html#sphx-glr-auto-examples-cluster-plot-mini-batch-kmeans-py
- https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html
- https://scikit-learn.org/stable/auto_examples/cluster/plot_kmeans_stability_low_dim_dense.html#sphx-glr-auto-examples-cluster-plot-kmeans-stability-low-dim-dense-py
- https://medium.com/analytics-vidhya/comparison-of-initialization-strategies-for-k-means-d5ddd8b0350e
